test/over_react_redux/hooks/use_selector_test.dart

```diff
@@ -0,0 +1,185 @@
+              (ConnectedCounter()..addTestId('sibling-counter'))(),
+            ), attachedToDocument: true);
+
+          expectModelCountValue(jacket, 0);
+          await clickSiblingConnectedModelCountIncrementButton(jacket);
+          expectModelCountValue(jacket, 1);
+        });
+
+        test('unless a custom equalityFn returns true', () async {
+          jacket = mount(
+            (ReduxProvider()..store = counterStore)(
+              (ModelCounterFn()..modelCountEqualityFn = (nextModel, prevModel) {
+                // Use 2 so that the equalityFn returns false once, and then returns true.
+                return nextModel.count == 2;
+              })(),
+              // Use a sibling connected component for dispatching actions in these tests
+              // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+              (ConnectedCounter()..addTestId('sibling-counter'))(),
+            ), attachedToDocument: true);
+
+          expectModelCountValue(jacket, 0);
+          await clickSiblingConnectedModelCountIncrementButton(jacket);
+          expectModelCountValue(jacket, 1, reason: 'Component should update when equalityFn returns false');
+          await clickSiblingConnectedModelCountIncrementButton(jacket);
+          expectModelCountValue(jacket, 1, reason: 'Component should not update when equalityFn returns true');
+        });
+      });
+    });
+
+    group('subscribes to custom context when createSelectorHook is used and redraws when', () {
+      group('the selector value is updated', () {
+        test('', () async {
+          jacket = mount(
+              (ReduxProvider()..store = counterStore)(
+                (ReduxProvider()
+                  ..store = bigCounterStore
+                  ..context = bigCounterContext
+                )(
+                  CustomContextCounterFn()(),
+                  // Use a sibling connected component for dispatching actions in these tests
+                  // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+                  (ConnectedBigCounter()..addTestId('big-sibling-counter'))(),
+                ),
+                // Use a sibling connected component for dispatching actions in these tests
+                // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+                (ConnectedCounter()..addTestId('sibling-counter'))(),
+              ), attachedToDocument: true);
+
+          expectBigCountValue(jacket, 9);
+          await clickSiblingConnectedBigCountIncrementButton(jacket);
+          expectBigCountValue(jacket, 109);
+
+          expectCountValue(jacket, 0);
+          await clickSiblingConnectedIncrementButton(jacket);
+          expectCountValue(jacket, 1);
+        });
+
+        test('unless a custom equalityFn returns false', () async {
```

Comment:
Should this be when the equality function is true?
```suggestion
        test('unless a custom equalityFn returns true', () async {
```